### PR TITLE
make load trigger fire and forget

### PR DESCRIPTION
**Pull request by 0xAustinWang**

### CODE REVIEW

**Review by mateusz-sekara — DISMISSED** [review]

File: integration-tests/load/ccip/helpers.go [hunk 1]
```diff
@@ -43,18 +45,137 @@ var (
...
+		"startblock", startBlock,
+	)
+
+	seqNums := make(map[testhelpers.SourceDestPair]SeqNumRange)
+	for _, cs := range otherChains {
+		seqNums[testhelpers.SourceDestPair{
+			SourceChainSelector: srcChainSel,
+			DestChainSelector:   cs,
+		}] = SeqNumRange{
+			Start: atomic.NewUint64(math.MaxUint64),
```
Comment (asoliman92) [inline integration-tests/load/ccip/helpers.go:82]: This always confuse me. Can we add a comment for why we're doing it like that?

File: integration-tests/load/ccip/helpers.go [hunk 2]
```diff
@@ -43,18 +45,137 @@ var (
...
+	endChannel := make(chan struct{})
+	// wait for load to finish + timeout duration to allow any stragglers to
+	go func() {
+		for {
+			select {
+			case <-loadFinished:
+				lggr.Infow("load finished, waiting before stopping transmit watcher",
+					"srcChain", srcChainSel)
+				go func() {
+					time.Sleep(tickerDuration)
```
Comment (asoliman92) [inline integration-tests/load/ccip/helpers.go]: Why sleep instead of using a ticker? Also what's the benefit of sleeping in a Go routine?

File: integration-tests/load/ccip/helpers.go [hunk 3]
```diff
@@ -43,18 +45,137 @@ var (
...
+			if event.SequenceNumber > seqNums[csPair].End.Load() {
+				seqNums[csPair].End.Store(event.SequenceNumber)
+			}
+		case <-ctx.Done():
+			lggr.Errorw("received context cancel signal for transmit watcher",
+				"srcChain", srcChainSel)
+			return
+		case <-endChannel:
+			for csPair, seqNums := range seqNums {
+				lggr.Debugw("pushing finalized sequence numbers for ",
```
Comment (asoliman92) [inline integration-tests/load/ccip/helpers.go]: Info instead of Debug?

File: integration-tests/load/ccip/helpers.go [hunk 4]
```diff
@@ -43,18 +45,137 @@ var (
...
+		}, sink, nil, nil)
+	})
+	defer subscription.Unsubscribe()
+
+	endChannel := make(chan struct{})
+	// wait for load to finish + timeout duration to allow any stragglers to
+	go func() {
+		for {
+			select {
+			case <-loadFinished:
```
Comment (asoliman92) [inline integration-tests/load/ccip/helpers.go]: I don't see where's this going to be triggered

File: integration-tests/load/ccip/helpers.go (hunk 1 above)
Comment (0xAustinWang) [reply integration-tests/load/ccip/helpers.go:82]: The initial values are set so that the first time we perform a comparison against a "real" value, we can update it. 

Ie if the first sequence number for this run is 100, it's lower than uint64 max so we can update it. 
Same for max sequence number

File: integration-tests/load/ccip/helpers.go (hunk 2 above)
Comment (0xAustinWang) [reply integration-tests/load/ccip/helpers.go]: Use a goroutine here to not block the main switch case statement. We want that to continue checking for transmits during this sleep time.

Using a ticker lets us check if load is done every cycle. If the end of the cycle corresponds with the end of the ticker, it's possible to miss a slow transmit

File: integration-tests/load/ccip/helpers.go (hunk 4 above)
Comment (0xAustinWang) [reply integration-tests/load/ccip/helpers.go]: https://github.com/smartcontractkit/chainlink/blob/9276b87cfcf1473b5edef966ef877b6c972e164a/integration-tests/load/ccip/ccip_test.go#L207

File: integration-tests/load/ccip/helpers.go (hunk 1 above)
Comment (asoliman92) [reply integration-tests/load/ccip/helpers.go:82]: Yeah. I noticed it down, writing a comment on this will be helpful for the reader.

File: integration-tests/load/ccip/helpers.go (hunk 2 above)
Comment (asoliman92) [reply integration-tests/load/ccip/helpers.go]: Why not just use the loadFinished instead of endChannel then in the loop below? and once you hit it wait for a tick and then do the logic.

Comment (0xAustinWang) [reply integration-tests/load/ccip/helpers.go]: I had that implementation initially, but doing it that way runs into an edge case where the ticker and loadFinished execute at the same time, which might miss a slow transmit. 

I took another look at it and added the time.Sleep one level up in ccip_test to make it simpler to understand

**Review by asoliman92 — APPROVED** [review]

**Review by mateusz-sekara — APPROVED** [review]
